dts/bindings/serial/uart-controller.yaml

```diff
@@ -10,12 +10,14 @@ properties:
     description: Clock frequency information for UART operation
   current-speed:
     type: int
+    default: 115200
```

Comment:
Lacking justification, see https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#rules-for-default-values

Reply:
Do you think that's true for all of the properties or just baudrate?
Also, Shouldn't it be a required property at least?

Reply:
All of them. If you add a default then they are required as the default will always be present unless something overrides

Reply:
What I was trying to propose in https://github.com/zephyrproject-rtos/zephyr/pull/83312#discussion_r1895600416 was to do this in the per-driver binding rather than the base binding.

As pointed out by @yishai1999 in https://github.com/zephyrproject-rtos/zephyr/pull/83312#discussion_r1895542662 there seems to be several drivers using `DT_INST_*_OR()` to have defaults in the driver code itself for these properties, e.g. here:
https://github.com/zephyrproject-rtos/zephyr/blob/6dce23326501a0850d2af9ae2af791ec0e7b6037/drivers/serial/uart_esp32.c#L1058-L1062

If there really *has* to be a default in these cases (which I'm not convinced of) having per-driver DT binding default values seems like the lesser of two evils, since then you can at least see the actual value that was used by looking at `build/zephyr/zephyr.dts`. However, either way, it'd be good to establish some consistent approach for this across all UART drivers.

Reply:
@nordicjm What do you think about it now?

Reply:
Where's the required description as per the above link stating where the default value comes from? E.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/sensor/vishay%2Cvcnl36825t.yaml#L47

Reply:
I guess we would have to ask the people who added the defaults in the drivers' code (@knthm , @bwitherspoon , @gohshunjing  + others for more drivers) but that fact is - They have defaults, they're just not written in the DTS which as @jhedberg said, makes it less easy to understand.